### ISSUE TITLE
plugin stats for bouncycastle-api are missing data.

### ISSUE THREAD

**Issue by jtnord** [jenkins-infra/helpdesk#4213]
### Service(s)

stats.jenkins.io

### Summary

The [statistics](https://stats.jenkins.io/pluginversions/bouncycastle-api.html) for the bouncycastle-api plugin are missing details of the last 4 CD releases.  
There is additionally a release present (`221.v8c725fc275c7`) that appears to be bogus.

### Reproduction steps

1. go to https://stats.jenkins.io/pluginversions/bouncycastle-api.html
2. check the version of the plugin in the top row
3. check the plugins [releases](https://github.com/jenkinsci/bouncycastle-api-plugin/releases) and [tags](https://github.com/jenkinsci/bouncycastle-api-plugin/tags)  (notice the last four releases and the lack of a `221.v8c725fc275c7` version
4. download https://updates.jenkins.io/current/plugin-versions.json and check the known versions of `bouncycastle-api`

Note that there are missing releases from the stats data and also release that is bogus.

**Comment by daniel-beck** [comment]
Versions appear in the HTML source, my guess is https://github.com/jenkins-infra/infra-statistics/commit/6d7432c852006f3b67a77f7b2aec98d0b94c9043 wasn't comprehensive enough and is missing legal punctuation.

**Comment by jtnord** [comment]
from @daniel-beck in a PM

> > completely bogus release
> We do not filter plugins by known public versions, and show what gets reported.

explains the `221.v8c725fc275c7` that has come from someone then :)

**Comment by daniel-beck** [comment]
> > show what gets reported

Noting that we could, like we do for [core](https://github.com/jenkins-infra/usage-log-decrypter/blob/master/src/main/java/org/jenkinsci/infra/log/PublicCores.java), we just don't bother at the moment.

**Comment by MarkEWaite** [comment]
Refer to a similar issue that was reported earlier.  At that time, it was decided that the results from the fix were worse than the issue:

* #3815

**Comment by daniel-beck** [comment]
So allowlisting public releases would be the way to go. Do we know enough about where `usage-log-decrypter` is run to ensure any such proposal would end up merged and used? Right now the repo state does not look good…

**Comment by dduportal** [comment]
Any objection if this issue moved to the repository (as there are no actionable for the Jenkins infra team)?

**Comment by mawinter69** [comment]
I assume this is related but for all plugins there is no update for the installation count. The last data is from June
e.g. 
![image](https://github.com/user-attachments/assets/5cf7f144-8d3b-4de3-986b-1f9e39f07280)
Is there some script not running that updates the stats or is it failing?

**Comment by timja** [comment]
Yes, after changes to require VPN Kohsuke's machine can't connect, @dduportal was in contact to resolve last I heard

**Comment by dduportal** [comment]
Absolutely: we are tracking the missing stats in https://github.com/jenkins-infra/helpdesk/issues/4285

**Comment by dduportal** [comment]
https://github.com/jenkins-infra/helpdesk/issues/4285 has been closed: stats until 29 Oct. have been integrated.

@jtnord @daniel-beck I still can't see the expected releases in the HTML but I'm not sure how it is generated. Are there additional work to do as per the discussion here?

**Comment by MarkEWaite** [comment]
I think that the [issue with "-" characters in the version string](https://github.com/jenkins-infra/helpdesk/issues/4213#issuecomment-2286454840) is still the reason that those releases are not displayed.

**Comment by daniel-beck** [comment]
> I think that

As confirmed in https://github.com/jenkins-infra/helpdesk/issues/4213#issuecomment-2273025837.

https://github.com/jenkins-infra/helpdesk/issues/4213#issuecomment-2286672473 still needs an answer.
